src/mission/mission.cpp

```diff
@@ -114,6 +157,29 @@ MissionPhase::MissionPhase(const ghoul::Dictionary& dictionary) {
             ));
         }
     }
+
+    if (p.actions.has_value()) {
+        _actions = p.actions.value_or(_actions);
+    }
+
+    if (p.milestones.has_value()) {
+        _milestones.reserve(p.milestones->size());
+        for (int i = 0; i < p.milestones->size(); i++) {
+            std::string name = p.milestones.value()[i].name;
+            Time newTime = Time(p.milestones.value()[i].date);
+            Milestone newDate = { name, newTime };
+            if (p.milestones.value()[i].description.has_value()) {
+                newDate.description = p.milestones.value()[i].description.value();
+            }
+            if (p.milestones.value()[i].image.has_value()) {
+                newDate.image = p.milestones.value()[i].image.value();
+            }
+            if (p.milestones.value()[i].link.has_value()) {
+                newDate.link = p.milestones.value()[i].link.value();
+            }
+            _milestones.emplace_back(newDate);
+        }
```

Comment:
Since you don't use the `i` parameter anywhere else, how about:
```suggestion
        for (const Milestone& milestone : *p.milestones) {
            std::string name = milestone.name;
            Time newTime = Time(milestone.date);
            Milestone newDate = { name, newTime };
            if (milestone.description.has_value()) {
                newDate.description = milestone.description.value();
            }
            if (milestone.image.has_value()) {
                newDate.image = milestone.image.value();
            }
            if (milestone.link.has_value()) {
                newDate.link = milestone.link.value();
            }
            _milestones.emplace_back(newDate);
        }
```

Reply:
And then one simplification more:
```cpp
newDate.description = milestone.description.value_or(newDate.description);
newDate.image = milestone.image.value_or(newDate.image);
newDate.link = milestone.link.value_or(newDate.link);
```

Reply:
I can't convert to Milestone from parameters - `no suitable user-defined conversion exists`

Reply:
Sorry; should be `Parameters::Milestone` as the type for the iterator variable, of course